tests/integ/test_large_query_breakdown.py

```diff
@@ -116,6 +116,77 @@ def test_no_valid_nodes_found(session, caplog):
     assert "Could not find a valid node for partitioning" in caplog.text
 
 
+def test_large_query_breakdown_external_cte_ref(session):
+    session._cte_optimization_enabled = True
+    if not session.sql_simplifier_enabled:
+        set_bounds(session, 50, 90)
+
+    base_select = session.sql("select 1 as A, 2 as B")
+    df1 = base_select.with_column("A", col("A") + lit(1))
+    df2 = base_select.with_column("B", col("B") + lit(1))
+    base_df = df1.union_all(df2)
+
+    df1 = base_df.with_column("A", col("A") + 1)
+    df2 = base_df.with_column("B", col("B") + 1)
+    for i in range(6):
+        df1 = df1.with_column("A", col("A") + i + col("A"))
+        df2 = df2.with_column("B", col("B") + i + col("B"))
+
+    df1 = df1.group_by("A").agg(sum_distinct(col("B")).alias("B"))
+    df2 = df2.group_by("B").agg(sum_distinct(col("A")).alias("A"))
+    final_df = df1.union_all(df2)
+
+    with SqlCounter(query_count=3, describe_count=0):
+        check_result_with_and_without_breakdown(session, final_df)
+
+    with patch.object(
+        session._conn._telemetry_client, "send_query_compilation_summary_telemetry"
+    ) as patch_send:
+        queries = final_df.queries
+
+    # assert that we did not break the plan
+    assert len(queries["queries"]) == 1
+
+    patch_send.assert_called_once()
+    _, kwargs = patch_send.call_args
+    summary_value = kwargs["compilation_stage_summary"]
+    assert summary_value["breakdown_failure_summary"] == [
+        {
```

Comment:
can we add a mocked telemetry test for summary for more than one plan to be more clear on the format

Reply:
I added it for another test below. for CTE where the breakdown is successful